### PR TITLE
IA-4051: Use PD section in Azure and GCP compute modal

### PR DESCRIPTION
**Pull request by lucymcnatt**

### CODE REVIEW

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js [hunk 1]
```diff
@@ -24,30 +23,32 @@ import {
...
 export const AzureComputeModalBase = ({
-  onDismiss, onSuccess, onError = onDismiss, workspace: { workspace: { namespace, name: workspaceName, workspaceId } }, runtimes, location, hideCloseButton = false
+  onDismiss, onSuccess, onError = onDismiss, workspace: { workspace: { namespace, name: workspaceName, workspaceId } }, runtimes, location, tool, hideCloseButton = false
 }) => {
   const [loading, setLoading] = useState(false)
   const [viewMode, setViewMode] = useState(undefined)
   const [currentRuntimeDetails, setCurrentRuntimeDetails] = useState(() => getCurrentRuntime(runtimes))
   const [computeConfig, setComputeConfig] = useState(defaultAzureComputeConfig)
-  const updateComputeConfig = (key, value) => setComputeConfig(_.set(key, value, computeConfig))
+  const updateComputeConfig = _.curry((key, value) => setComputeConfig(_.set(key, value)))
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js:38]: I'm planning to change this in the future... It's wild that we're doing anything else here.
Really what we SHOULD be doing is whenever we want to update compute config is the following:
```
setComputeConfig({
    ...computeConfig,
    [key]: value
})
```

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js [hunk 2]
```diff
@@ -204,9 +185,9 @@ export const AzureComputeModalBase = ({
       [viewMode === 'deleteEnvironment',
         () => Ajax().Runtimes.runtimeV2(workspaceId, currentRuntimeDetails.runtimeName).delete()], //delete runtime
       [Utils.DEFAULT, () => {
+        // TODO: We DO currently support re-attaching azure disks
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js]: Do we have any ticket number we can add to this TODO and the next TODO on 201?

File: src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts [hunk 3]
```diff
@@ -155,5 +154,17 @@ describe('compute-modal-component', () => {
       // Assert
       expect(updateComputeConfig).toBeCalledWith({ displayName: 'Balanced', label: 'pd-balanced', regionToPricesName: 'monthlyBalancedDiskPrice' })
     })
+
+    it('should call updateComputeConfig with proper value on changing size', async () => {
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts:157]: Could we add a little more info here? It's confusing that it's being called with 500 when the user types 0.

It makes sense that it starts at 50 and then typing 0 would make it 500. Just wondering if we can clarify that.

File: src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.ts [hunk 4]
```diff
@@ -1,48 +1,76 @@
...
+  tool: any
+  onDismiss: () => void
+}
+
+export const handleLearnMoreAboutPersistentDisk = ({ setViewMode }) => {
+  setViewMode('aboutPersistentDisk')
+  Ajax().Metrics.captureEvent(Events.aboutPersistentDiskView)
+}
+
+// TODO, check if unattached azure PD
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.ts]: Do we have any ticket for this?

File: src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts (hunk 3 above)
Comment (lucymcnatt) [reply src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts:157]: ohhh yeah I was missing the `userEvent.clear()` method for this one, updated

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js (hunk 2 above)
Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js]: NIT: If there's nothing to do can we remove the TODO part? :)

File: src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts [hunk 5]
```diff
@@ -20,16 +20,17 @@ import {
...
 
 
 const defaultIComputeConfig = {
-  selectedPersistentDiskSize: defaultGcePersistentDiskSize,
-  selectedPersistentDiskType: defaultPersistentDiskType,
+  persistentDiskSize: defaultGcePersistentDiskSize,
+  persistentDiskType: defaultPersistentDiskType,
   masterMachineType: getDefaultMachineType(false, toolLabels.RStudio),
   masterDiskSize: defaultDataprocMasterDiskSize,
+  diskSize: 0, //TODO: set to a default disk size for Azure
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts]: 50 seems to be our default

File: src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js (hunk 2 above)
Comment (lucymcnatt) [reply src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js]: well I think this is what [Jake's ticket](https://broadworkbench.atlassian.net/browse/IA-4052) addresses, so I added that ticket # here

**Review by n8landolt — APPROVED** [review]

File: src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts (hunk 5 above)
Comment (lucymcnatt) [reply src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.test.ts]: realized this was leftover from before we changed diskSize --> persistentDiskSize, so I nixed it

**Pull request merged**